### PR TITLE
feat: [CDS-46165]: UI Add Human Readable Plan support in NG

### PR DESCRIPTION
**Pull request by akhilpandey1**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by akhilpandey1** [comment]
retrigger cypress-rest, retrigger cypress-pipeline, retrigger cypress-cv

**Comment by akhilpandey1** [comment]
retrigger cypress-rest

**Comment by akhilpandey1** [comment]
retrigger cypress-cv

**Comment by akhilpandey1** [comment]
retrigger cypress-pipeline

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/TerraformInterfaces.ts [hunk 1]
```diff
@@ -468,6 +469,12 @@ export const onSubmitTFPlanData = (values: any): TFPlanFormData => {
       : false
   }
 
+  if (values?.spec?.configuration?.exportTerraformHumanReadablePlan) {
+    configObject['exportTerraformHumanReadablePlan'] = values?.spec?.configuration?.exportTerraformHumanReadablePlan
+      ? values?.spec?.configuration?.exportTerraformHumanReadablePlan
+      : false
+  }
+
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/TerraformInterfaces.ts:477]: Inside the if block, `values?.spec?.configuration?.exportTerraformHumanReadablePlan` will always be true, so you can use `configObject['exportTerraformHumanReadablePlan'] = true`.

or maybe remove the if block:

```suggestion
    configObject['exportTerraformHumanReadablePlan'] = values?.spec?.configuration?.exportTerraformHumanReadablePlan ?? false
```

**Review by rameshx — COMMENTED** [review]
Make sure that the changes in `src/services/cd-ng/index.tsx` are intended.

File: src/services/cd-ng/index.tsx [hunk 2]
```diff
@@ -14301,7 +14294,7 @@ export interface TerraformBackendConfig {
 }
 
 export interface TerraformBackendConfigSpec {
-  type?: string
+  [key: string]: any
```
Comment (vikrant-harness) [inline src/services/cd-ng/index.tsx:14297]: @akhilpandey1 was this manually updated? I took the swagger from the pre-qa and this change is present there

Comment (akhilpandey1) [reply src/services/cd-ng/index.tsx:14297]: @vikrant-harness Yes, Its manually updated, without this the Jest PR checks are failing

Comment (vikrant-harness) [reply src/services/cd-ng/index.tsx:14297]: We never update this file manually , This is auto generated from swagger.

Comment (vikrant-harness) [reply src/services/cd-ng/index.tsx:14297]: The sync with BE and FE changes if we manually update this

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/TerraformInterfaces.ts (hunk 1 above)
Comment (akhilpandey1) [reply src/modules/75-cd/components/PipelineSteps/Common/Terraform/TerraformInterfaces.ts:477]: @rameshx I inferred the block just checks if the values?.spec?.configuration?.exportTerraformHumanReadablePlan is not null?
and thus inside it, the value can be both false/true, 
also,
Its done the same way for exportTerraformPlanJson if you expand the above lines

**Comment by akhilpandey1** [comment]
retrigger jest

**Comment by akhilpandey1** [comment]
retrigger coverage

**Review by vikrant-harness — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Review by Tamaraius — APPROVED** [review]

**Pull request merged**